### ISSUE TITLE
Can't start Vagrant VM - "NVMe configuration error" - Windows 10 Host

### ISSUE THREAD

**Issue by kvietmeier** [spdk/spdk#292]
Please use the issue tracker only for reporting suspected issues.

See [The SPDK Community Page](http://www.spdk.io/community/) for other SPDK communications channels.

## Expected Behavior
The VM should start after "vagrant up"

## Current Behavior
VM fails to start.   It is created in Virtualbox but can't start with ter following error:

There was an error while executing `VBoxManage`, a CLI used by Vagrant
for controlling VirtualBox. The command and stderr is shown below.

Command: ["startvm", "eb13591a-5f39-4520-b0a0-f4601a315cf1", "--type", "headless"]

Stderr: VBoxManage.exe: error: NVMe configuration error: LUN#0: Failed to set I/O request size! (VERR_INVALID_PARAMETER).
VBoxManage.exe: error: NVMe initialisation error: failed to initialize a namespace (VERR_INVALID_PARAMETER)
VBoxManage.exe: error: Details: code E_FAIL (0x80004005), component ConsoleWrap, interface IConsole

## Possible Solution
Possible issue with staertup script in the VM?   Need to modify the Packer billd sertup.

## Steps to Reproduce

1.   git clone repo
2.   run vagrant up
3.
4.

## Context (Environment including OS version, SPDK version, etc.)
Virtualbox - 5.2.8 r121009 (Qt5.6.2)
Extension plugin - 5.2.10
Windows 10 - v1607   Build 14393.2189
puppetlabs/ubuntu-16.04-64-nocm (virtualbox, 1.0.0)

PS C:\Users\ksvietme\Documents\Projects\Intel\spdk> cat .\CHANGELOG.md
# Changelog

**## v18.04: (Upcoming Release)**

**Comment by peluse** [comment]
I've got vbox on my Windows10 machine at home, I'll see if I can reproduce this either today or over the weekend.  Feel free to ping me if you don't hear back from me by Mon :)

**Comment by kvietmeier** [comment]
Update - I added the NVMe dedvice lineds to a generic Centos Vagrtatfile - same problem so this looks like a Virtuabox issue.

**Comment by wanqunintel** [comment]
@zongwenqian, wenqian is also trying to reproduce it, but she need to setup the environment first and then see what happens there in our lab.

**Comment by johnmeneghini** [comment]
Paul, you did this on a MacBook?   I’m running OSX 10.12.6.  I already had Vbox installed, so I installed the Vbox extensions and then vagrant.

Do I really need to install cygwin?    I was hoping to run vagrant from my OSX terminal window.

However, after installing everything,  I can’t find the vagrant utility.

bash-3.2$ whereis vagrant
bash-3.2$ echo $PATH
/opt/local/bin:/opt/local/sbin:~/bin:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin:/opt/X11/bin:/opt/X11/bin

/John

**Comment by peluse** [comment]
Hi John-

This github issue is about running on Windows. I think I replied to the dist list for your question on running on OS/X.  I did both.  On Windows no you don't need cygwin but I can't stand the windows command prompt so did everything from a cygwin window.  

On OS/X I think I provided more detail but my vagrant installed to usr/local/bin/vagrant Maybe just resintall the latest vbox and vagrant and try again?

**Comment by yidong0635** [comment]
![image](https://user-images.githubusercontent.com/39082016/40463893-789e5b4a-5f4a-11e8-9864-98c27bf0c96e.png)

**Comment by yidong0635** [comment]
![image](https://user-images.githubusercontent.com/39082016/40463993-fdb490ce-5f4a-11e8-9e66-42eb4f580849.png)

**Comment by wanqunintel** [comment]
@kvietmeier as Paul and Yidong has setup and run vagrant successfully, maybe we can close it

**Comment by yidong0635** [comment]
@kvietmeier  I  simulated and understood your error, you should install  virtualbox extension pack:
it should meet your virtualbox version. 
such as -- https://www.downloadcrew.co.uk/article/20279-virtualbox_extension_pack
`vboxmanage extpack install Oracle_VM_VirtualBo                                                                                                         x_Extension_Pack-5.2.12.vbox-extpack`

**Comment by yidong0635** [comment]
It seems to the virtualbox issue.
![image](https://user-images.githubusercontent.com/39082016/41456051-e9755bde-70b1-11e8-86be-056c2b9c2a63.png)

**Comment by kvietmeier** [comment]
Yidong is using Linux/OS X - not applicable

It sounds like Paul reinstalled Windows?   That is not a viable resolution - no one is going to reinstall their OS to get this working.

This looks very much like a Virtuabox for Windows issue. 

I am running the latest Windows release and patch set that my IT org will allow.
 VBox 5.2.8 r121009 (Qt5.6.2)
 Host extensions - the latest as of 2 weeks ago
 Vagrant 2.0.1 - slightly down reved.   

Still can't start the VM.   But I agree with Yidong, this is a Vbox issue.   Happy to take this offline to resolve directly - ping me on Skype.

And Paul - there are some very good terminal solutions for Windows 10 - checkout ConEmu, I do everything in a PowerShell terminal session.  Think of me as the advocate for all of those admins out there using PowerShell.

**Comment by kvietmeier** [comment]
**Resolved -** 
I went back and  double checked all versions and updated to the latest possible.  I can start the VM.

No need to reinstall Windows - but I did need the latest Extension Pack which has an NVMe update.
Versions:
VBox - 5.2.12 r122591 (Qt5.6.2)
EPack - 5.2.12
Vagrant - 2.0.1
Windows - Holiday Edition

![image](https://user-images.githubusercontent.com/9323750/41548182-00cee67a-72d7-11e8-847f-9e9ad0d45aef.png)
